### DISCUSSION TITLE
Modal as confirm dialog

### DISCUSSION THREAD

**Discussion by ev3rlost** [wire-elements/modal#390] [Q&A]
Is it possible to have the modal behave like a confirm dialog?

I came across the problem, where I display the modal, have two buttons in it (cancel, delete), click on delete (runs a method delete inside the component), deletes the record, closes the modal - but then I get a 404 not found.

```
// posts.blade.php

<button wire:click="$dispatch('openModal', { component: 'delete', arguments: { post: {{ $post->id }} }})">
```

```
// delete.blade.php

<div>Are you sure?</div>
<button wire:click="delete">Confirm delete<button>
```

```
// Delete.php

public Post $post

public function delete()
{
    $this->authorize('delete', $this->post);
    $this->post->delete();
    $this->closeModal();
}
```

Am I missing something?

**Comment by SmoKE585** [comment]
same thing for me :(

**Comment by lrljoe** [comment]
If you're using Livewire 3, then use the new wire:confirm approach instead to confirm, then run your delete authorization, and close the modal.  Take note of the order, and customisations of closing the modal etc

If you are insistent on doing it another way, then you may end up deleting the record for the page you're on, and you'll get a 404 unless you force a redirect, or catch a model not found correctly.

Don't use this for confirmation modals, it's not efficient for that as you're making more db calls than you need to
